13-multiplecomponents-lab/src/03-shoppinglist-filters.jsx

```diff
@@ -1,7 +1,170 @@
-/**
- * Lab 03: Shopping List Filters
- *
- * Add two filters to the shopping list built in previous lab:
- *  - Ability to show only products matching a given name
- *  - Ability to show only products that were fully bought
- */
+/**
+ * Lab 03: Shopping List Filters
+ *
+ * Add two filters to the shopping list built in previous lab:
+ *  - Ability to show only products matching a given name
+ *  - Ability to show only products that were fully bought
+ */
+import React from "react";
+
+function ShoppingItemsCount(props) {
+	return <p>Total items: {props.count}</p>
+}
+
+export default class ShoppingListFilter extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			items: [],
+		};
+		this.getParameters.bind(this);
+	}
+
+	addItem() {
+		const nameInput = this.refs.newItemName;
+		const imageInput = this.refs.newItemImage;
+		const neededQuantityInput = this.refs.newItemNeededQuantity;
+		const id = new Date().getTime();
+		const items = this.state.items;
+		items.push({id: id, name: nameInput.value, image: imageInput.value, neededQuantity: neededQuantityInput.value});
+		this.setState(items);
+
+		nameInput.value = '';

```

Comment:
👍 Totally agree with you, but:
1. Saw it in class :)
2. From previous code-review on the time converter, I got the impression that we shouldn't put redundant variables on the state unless we really need them.
Again, totally agree with you, but i'd be happy to know what is better in terms of best practices